### PR TITLE
Report API v2

### PR DESCRIPTION
**Pull request by dss-vipps**
Links to the important full docs of this PR:

Guide: https://github.com/vippsas/vipps-report-api/blob/v2/api-guide/settlement.md

Swagger: https://github.com/vippsas/vipps-report-api/blob/v2/docs/swagger.yaml

Changelog: https://github.com/vippsas/vipps-report-api/blob/v2/CHANGELOG.md

## About timing for merging this

The implementation of the API is pretty close to being done, so it is probably simplest to make the v2 available before we publish the docs, so we don't have to explain why API isn't there when the docs are merged. Both versions will be available for some time.

### CODE REVIEW

**Review by cloveras — APPROVED** [review]
Looks good! We can add some final polishing if needed, but this would be great to merge ASAP, since we ned it both internally and externally.

File: docs/swagger.yaml [hunk 1]
```diff
@@ -105,88 +112,128 @@ paths:
...
                   }
+        default:
+          description: Default problem response
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/Problem"
 
-  "/report/v1/ledgertransactions":
+  "/report/v2/ledgers/{LedgerId}/{Account}/dates/{LedgerDate}":
```
Comment (olillevik) [inline docs/swagger.yaml:122]: The {Account} part of this url does not look right.

File: docs/swagger.yaml [hunk 2]
```diff
@@ -105,88 +112,128 @@ paths:
...
+                $ref: "#/components/schemas/Problem"
+
+  "/report/v2/ledgers/{LedgerId}/{Account}/feed":
+    get:
+      tags:
+        - /report/v2/ledgers
+      summary: Fetch report by feed
+      description: |-
+        Continuous stream of data. Upon reaching the end of the feed the `cursor` will not become empty and `items` can 
+        become en empty list. One should retry to fetch the same `cursor` until new data and a new `cursor` will be returned.
```
Comment (olillevik) [inline docs/swagger.yaml:193]: Might just be a spelling error but I am not sure what is meant here. "en empty list"

**Review by olillevik — CHANGES_REQUESTED** [review]

File: docs/swagger.yaml (hunk 1 above)
Comment (cloveras) [reply docs/swagger.yaml:122]: We should use camelCase for path parameters, as specified in the RFC: https://github.com/vippsas/rfc/blob/master/docs/0001-api-guidelines.md#use-camelcase

**Review by cloveras — CHANGES_REQUESTED** [review]
See comments.

File: docs/swagger.yaml (hunk 2 above)
Comment (rebekaburnett) [inline docs/swagger.yaml:193]:
```suggestion
        become an empty list. One should retry to fetch the same `cursor` until new data and a new `cursor` will be returned.
```

File: docs/swagger.yaml [hunk 3]
```diff
@@ -105,88 +112,128 @@ paths:
...
+              schema:
+                $ref: "#/components/schemas/Problem"
+
+  "/report/v2/ledgers/{LedgerId}/{Account}/feed":
+    get:
+      tags:
+        - /report/v2/ledgers
+      summary: Fetch report by feed
+      description: |-
+        Continuous stream of data. Upon reaching the end of the feed the `cursor` will not become empty and `items` can
```
Comment (rebekaburnett) [inline docs/swagger.yaml:192]:
```suggestion
        Continuous stream of data. Upon reaching the end of the feed, the `cursor` will not become empty and `items` can 
```

**Review by rebekaburnett — DISMISSED** [review]

**Review by cloveras — DISMISSED** [review]

**Review by olillevik — DISMISSED** [review]

**Review by cloveras — APPROVED** [review]

**Pull request merged**